src/category.rs

```diff
@@ -0,0 +1,164 @@
+use std::collections::HashSet;
+use time::Timespec;
+
+use conduit::{Request, Response};
+use conduit_router::RequestParams;
+use pg::GenericConnection;
+use pg::rows::Row;
+
+use {Model, Crate};
+use db::RequestTransaction;
+use util::{RequestUtils, CargoResult, ChainError};
+use util::errors::NotFound;
+
+#[derive(Clone)]
+pub struct Category {
+    pub id: i32,
+    pub category: String,
+    pub created_at: Timespec,
+    pub crates_cnt: i32,
+}
+
+#[derive(RustcEncodable, RustcDecodable)]
+pub struct EncodableCategory {
+    pub id: String,
+    pub category: String,
+    pub created_at: String,
+    pub crates_cnt: i32,
+}
+
+impl Category {
+    pub fn find_by_category(conn: &GenericConnection, name: &str)
+                            -> CargoResult<Option<Category>> {
+        let stmt = try!(conn.prepare("SELECT * FROM categories \
+                                      WHERE category = $1"));
+        let rows = try!(stmt.query(&[&name]));
+        Ok(rows.iter().next().map(|r| Model::from_row(&r)))
+    }
+
+    pub fn find_all_by_category(conn: &GenericConnection, names: &[String])
+                                  -> CargoResult<Vec<Category>> {
+        let stmt = try!(conn.prepare("SELECT * FROM categories \
+                                      WHERE category = ANY($1)"));
+        let rows = try!(stmt.query(&[&names]));
+        Ok(rows.iter().map(|r| Model::from_row(&r)).collect())
+    }
+
+    pub fn encodable(self) -> EncodableCategory {
+        let Category { id: _, crates_cnt, category, created_at } = self;
+        EncodableCategory {
+            id: category.clone(),
```

Comment:
Sure! Should categories.txt still list only the title-cased, punctuated version, and then the slug is lowercased, replace all spaces with hyphens, and remove other punctuation? Or should categories.txt list the slug as well?

If people specify the slug as their crate's category value, should that be valid?